### PR TITLE
feat(ls): provide OpenAPI 3.1.0 lint rules for Security Scheme object

### PR DESCRIPTION
**Pull request by tim-lai**
### Description

* lint rules for OpenAPI 3.1.0 Security Scheme Object
* completion rules for OpenAPI 3.1.0 Security Scheme Object
* add `targetSpecs` for OAS3.0.x `/openapi/security-scheme/lint/type--equals-3-0.ts`

### Motivation and Context

Ref: #2061 
Ref: #2062 

### My PR contains...

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-lai** [comment]
@char0n I'm not seeing the difference between the OAS3.0 and OAS3.1 specs of why there needs to be separate `openapi/security-scheme/lint/allowed-fields-3-0.ts` and `openapi/security-scheme/lint/allowed-fields-3-1.ts`. So they both currently exist in this PR. I also don't see the reference to `$ref`, but see that it was added as part of the `Reference Object` update.

File: packages/apidom-ls/src/config/openapi/security-scheme/lint/type--equals-3-0.ts [hunk 1]
```diff
@@ -11,6 +11,12 @@ const typeEquals3_0Lint: LinterMeta = {
   linterParams: [['apiKey', 'http', 'oauth2', 'openIdConnect']],
   marker: 'value',
   target: 'type',
+  targetSpecs: [
```
Comment (char0n) [inline packages/apidom-ls/src/config/openapi/security-scheme/lint/type--equals-3-0.ts:14]: good catch, thanks

**Comment by char0n** [comment]
> @char0n I'm not seeing the difference between the OAS3.0 and OAS3.1 specs of why there needs to be separate `openapi/security-scheme/lint/allowed-fields-3-0.ts` and `openapi/security-scheme/lint/allowed-fields-3-1.ts`. So they both currently exist in this PR. I also don't see the reference to `$ref`, but see that it was added as part of the `Reference Object` update.

You can find the answer here: https://github.com/swagger-api/apidom/commit/d476aeffe0fe1474646dcfd4de1bb1fd39d8f151#r87246050

File: packages/apidom-ls/src/config/openapi/security-scheme/lint/allowed-fields-3-1.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    [
+      'type',
+      'description',
+      'name',
+      'in',
+      'scheme',
+      'bearerFormat',
+      'flows',
+      'openIdConnectUrl',
+      '$ref',
```
Comment (char0n) [inline packages/apidom-ls/src/config/openapi/security-scheme/lint/allowed-fields-3-1.ts:21]:
```suggestion
```

File: packages/apidom-ls/src/config/openapi/security-scheme/lint/allowed-fields-3-1.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ],
+    'x-',
+  ],
+  marker: 'key',
```
Comment (char0n) [inline packages/apidom-ls/src/config/openapi/security-scheme/lint/allowed-fields-3-1.ts:25]:
```suggestion
  marker: 'key',
    conditions: [
    {
      function: 'missingField',
      params: ['$ref'],
    },
  ],
```

File: packages/apidom-ls/src/config/openapi/security-scheme/lint/allowed-fields-3-1.ts (hunk 2 above)
Comment (char0n) [reply packages/apidom-ls/src/config/openapi/security-scheme/lint/allowed-fields-3-1.ts:21]: Explanation for the change in https://github.com/swagger-api/apidom/commit/d476aeffe0fe1474646dcfd4de1bb1fd39d8f151#r87246050

File: packages/apidom-ls/src/config/codes.ts [hunk 4]
```diff
@@ -848,6 +848,8 @@ enum ApilintCodes {
   OPENAPI3_1_REFERENCE = 7040000,
   OPENAPI3_1_REFERENCE_FIELD_$REF_ALLOWED_SIBLINGS = 7040100,
 
+  OPENAPI3_1_SECURITY_SCHEME_FIELD_TYPE_EQUALS = 7240100,
```
[the remark this answers is not in the document]
Comment (tim-lai) [reply packages/apidom-ls/src/config/codes.ts]: ok for now. change made.

**Comment by tim-lai** [comment]
> You can find the answer here

Interesting read, a lot of backstory going back quite a ways there. 😄

**Review by char0n — APPROVED** [review]
